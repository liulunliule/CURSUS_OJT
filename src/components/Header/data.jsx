import axios from "axios";
export const fetchUserMessage = async () => {
    try {
      const response = await axios.get("https://6694e50b4bd61d8314c916af.mockapi.io/Message");
      return response.data;
    } catch (error) {
      console.error("Error fetching user posts:", error);
      throw error;
    }
  };
  export const fetchUserNotification = async () => {
    try {
      const response = await axios.get("https://66751909a8d2b4d072eeb172.mockapi.io/notification");
      return response.data;
    } catch (error) {
      console.error("Error fetching user posts:", error);
      throw error;
    }
  };