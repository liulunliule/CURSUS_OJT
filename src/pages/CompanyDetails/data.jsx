import { useState, useEffect } from "react";
import axios from "axios";

export const useFetchOffices = () => {
  const [offices, setOffices] = useState([]);

  useEffect(() => {
    axios.get("https://668f4a0880b313ba09178dee.mockapi.io/api/offices")
      .then(response => {
        setOffices(response.data);
      })
      .catch(error => {
        console.error("Error fetching office data:", error);
      });
  }, []);

  return offices;
};