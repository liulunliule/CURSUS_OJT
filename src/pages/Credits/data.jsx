import { useState, useEffect } from "react";
import axios from "axios";
import {
    fetchCreditData,
    fetchTransactionsData,
} from "../../services/apiService";

export const useFetchCreditsData = () => {
    const [addedCredits, setAddedCredits] = useState({});
    const [transactions, setTransactions] = useState([]);

    useEffect(() => {
        const fetchAddedCredits = async () => {
            try {
                const response = await fetchCreditData();
                setAddedCredits(response.data[0]);
            } catch (error) {
                console.error("Error fetching added credits data:", error);
            }
        };

        const fetchTransactions = async () => {
            try {
                const response = await fetchTransactionsData();
                setTransactions(response.data);
            } catch (error) {
                console.error("Error fetching transactions data:", error);
            }
        };

        fetchAddedCredits();
        fetchTransactions();
    }, []);

    return { addedCredits, transactions };
};
