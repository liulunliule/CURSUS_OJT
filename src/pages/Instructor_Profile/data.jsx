import axios from "axios";

export const fetchUserInfo = async (targetId) => {
  try {
    const response = await axios.get("https://66751909a8d2b4d072eeb172.mockapi.io/InstructorProfile");
    return response.data.find((item) => item.id === targetId);
  } catch (error) {
    console.error("Error fetching user info:", error);
    throw error;
  }
};

export const fetchUserPosts = async () => {
  try {
    const response = await axios.get("https://6677cf6e0bd45250561c9488.mockapi.io/course");
    return response.data;
  } catch (error) {
    console.error("Error fetching user posts:", error);
    throw error;
  }
};

export const fetchUserReviews = async () => {
  try {
    const response = await axios.get("https://6677cf6e0bd45250561c9488.mockapi.io/reviews");
    return response.data;
  } catch (error) {
    console.error("Error fetching user reviews:", error);
    throw error;
  }
};

export const fetchSubscriptions = async () => {
  try {
    const response = await axios.get("https://6678f8ae0bd4525056207d59.mockapi.io/subscriptions");
    return response.data;
  } catch (error) {
    console.error("Error fetching subscriptions:", error);
    throw error;
  }
};
