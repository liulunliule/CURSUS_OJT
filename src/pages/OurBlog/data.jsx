import { useState, useEffect } from "react";
import axios from "axios";

export const useFetchBlogs = () => {
  const [blogs, setBlogs] = useState([]);

  useEffect(() => {
    axios.get("https://668f4a0880b313ba09178dee.mockapi.io/api/blogs")
      .then((response) => setBlogs(response.data))
      .catch((error) => console.error("Error fetching blog data:", error));
  }, []);

  return blogs;
};