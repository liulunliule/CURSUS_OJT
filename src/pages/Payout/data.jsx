import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchNextPayout, fetchTransactions } from '../../redux/features/payoutSlice';

export const useFetchPayoutData = () => {
  const dispatch = useDispatch();
  const nextPayout = useSelector((state) => state.payout.nextPayout);
  const transactions = useSelector((state) => state.payout.transactions);
  const status = useSelector((state) => state.payout.status);
  const error = useSelector((state) => state.payout.error);

  useEffect(() => {
    if (status === 'idle') {
      dispatch(fetchNextPayout());
      dispatch(fetchTransactions());
    }
  }, [status, dispatch]);

  return { nextPayout, transactions, status, error };
};