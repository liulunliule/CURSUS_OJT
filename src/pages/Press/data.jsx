import { useState, useEffect } from "react";
import axios from "axios";

export const useFetchPressData = () => {
  const [news, setNews] = useState([]);
  const [releases, setReleases] = useState([]);

  useEffect(() => {
    const fetchNews = async () => {
      try {
        const response = await axios.get('https://6690833cc0a7969efd9c4da4.mockapi.io/api/press_news');
        setNews(response.data);
      } catch (error) {
        console.error("Error fetching news data:", error);
      }
    };

    const fetchReleases = async () => {
      try {
        const response = await axios.get('https://6690833cc0a7969efd9c4da4.mockapi.io/api/press_releases');
        setReleases(response.data);
      } catch (error) {
        console.error("Error fetching releases data:", error);
      }
    };

    fetchNews();
    fetchReleases();
  }, []);

  return { news, releases };
};