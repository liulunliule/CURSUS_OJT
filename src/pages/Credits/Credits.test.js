import '@testing-library/jest-dom/extend-expect';
import { useFetchCreditsData } from './data';

jest.mock('./data', () => ({
  useFetchCreditsData: jest.fn(),
}));

describe('Credits Component', () => {
  const mockData = {
    addedCredits: { amount: '$100', method: 'Credit Card', description: 'purchases' },
    transactions: [
      { amount: '$100', method: 'Credit Card', date: '2023-07-27' },
      { amount: '$50', method: 'PayPal', date: '2023-07-28' },
    ],
    status: 'succeeded',
    error: null,
  };

  it('renders loading state', () => {
    useFetchCreditsData.mockReturnValue({ status: 'loading' });
  });

  it('renders error state', () => {
    useFetchCreditsData.mockReturnValue({ ...mockData, status: 'failed', error: 'Failed to fetch data' });
  });

  it('renders credits data correctly', () => {
    useFetchCreditsData.mockReturnValue(mockData);
  });

  it('applies active class when isShowAll is true', () => {
    useFetchCreditsData.mockReturnValue(mockData);
  });
});