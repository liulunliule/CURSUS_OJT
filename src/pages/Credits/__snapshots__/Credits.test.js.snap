// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Credits 1`] = `
<div
  className="credits-page "
>
  <div
    className="content"
  >
    <div
      className="credits-header"
    >
      <h2>
        Credits
      </h2>
    </div>
    <div
      className="credits-content"
    >
      <div
        className="added-credits"
      >
        <h3>
          Added Credits
        </h3>
        <h6 />
        <h5>
          via 
        </h5>
        <p>
          Your credits will be used on 
          <span />
        </p>
      </div>
      <div
        className="saved-cards"
      >
        <h3>
          Saved Cards
        </h3>
        <div
          className="card"
        >
          <img
            alt="MasterCard"
            src="MasterCard.jpg"
          />
          <p>
            Added: 25 Mar 2020
          </p>
          <button>
            Change Card
          </button>
        </div>
      </div>
      <div
        className="add-credit-balance"
      >
        <h3>
          Add Credit Balance
        </h3>
        <form>
          <div
            className="form-row"
          >
            <label>
              Add Balance
              <input
                placeholder="$0"
                type="text"
              />
            </label>
          </div>
          <div
            className="form-row"
          >
            <label>
              Holder Name
              <input
                placeholder="Enter Holder Name"
                type="text"
              />
            </label>
            <label>
              Card Number
              <input
                placeholder="Card #"
                type="text"
              />
            </label>
          </div>
          <div
            className="form-row"
          >
            <label>
              Expiration Month
              <select>
                <option>
                  Month
                </option>
                <option>
                  January
                </option>
                <option>
                  February
                </option>
                <option>
                  March
                </option>
                <option>
                  April
                </option>
                <option>
                  May
                </option>
                <option>
                  June
                </option>
                <option>
                  July
                </option>
                <option>
                  August
                </option>
                <option>
                  September
                </option>
                <option>
                  October
                </option>
                <option>
                  November
                </option>
                <option>
                  December
                </option>
              </select>
            </label>
            <label>
              Expiration Year
              <input
                placeholder="Year"
                type="text"
              />
            </label>
            <label>
              Expiration Year
              <input
                placeholder="CVC"
                type="text"
              />
            </label>
          </div>
          <button
            type="submit"
          >
            Add Credit
          </button>
        </form>
      </div>
    </div>
    <div
      className="transactions"
    >
      <table>
        <thead>
          <tr>
            <th>
              Amount
            </th>
            <th>
              Credit Method
            </th>
            <th>
              Date Processed
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</div>
`;